### PR TITLE
Cc/preprocessor

### PR DESCRIPTION
**Pull request by Supercip971**

### CODE REVIEW

File: sources/libs/cc/ast/decl.h [hunk 1]
```diff
@@ -9,6 +9,7 @@ typedef enum
 {
     CDECL_NIL,
 
+    CDECL_EMPTY,
```
Comment (sleepy-monax) [inline sources/libs/cc/ast/decl.h:12]: What does an empty declaration look like?

File: sources/libs/cc/parse/lexer.c [hunk 2]
```diff
@@ -6,6 +6,15 @@ struct
     LexemeType type;
     char const *literal;
 } keywords[] = {
+    {CLEX_PRAGMA, "#pragma"},
+    {CLEX_DEFINE, "#define"},
```
Comment (sleepy-monax) [inline sources/libs/cc/parse/lexer.c:10]: `#                 define HELLO` is valid in C

File: sources/libs/cc/proc/macro.c [hunk 3]
```diff
@@ -0,0 +1,163 @@
...
+
+void cproc_parse_define(Lex *source, CProcContext *ctx)
+{
+    CProcMacro macro = {};
+    vec_init(&macro.args, ctx->alloc);
+    vec_init(&macro.source, ctx->alloc);
+
+    cparse_skip_separator(source, CLEX_DEFINE);
+
+    if (lex_curr(source).type != CLEX_IDENT)
```
Comment (sleepy-monax) [inline sources/libs/cc/proc/macro.c:13]: Any token can be redefined in C like: `#define + -` ;)

File: sources/libs/cc/ast/decl.h (hunk 1 above)
Comment (Supercip971) [reply sources/libs/cc/ast/decl.h:12]: for the moment it's just:
int i; <= variable declaration
; <= empty declaration

File: sources/libs/cc/proc/macro.c (hunk 3 above)
Comment (Supercip971) [reply sources/libs/cc/proc/macro.c:13]: oh crap

**Review by sleepy-monax — CHANGES_REQUESTED** [review]
Great contribution but lack some unit tests

**Pull request merged**